src/key-queue.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * The KeyQueue in Tram-One is a basic list of keys
+ * that needs to be persisted in the globalSpace.
+ *
+ * Currently this is used with useStore to keep track of what
+ * stores need to be associated with generated elements
+ */
+
+import { buildNamespace } from './namespace';
+
+const newDefaultKeyQueue = () => {
+	return [] as string[];
+};
+
+export const { setup: setupKeyQueue, get: getKeyQueue, set: setKeyQueue } = buildNamespace(newDefaultKeyQueue);
+
+/**
+ * clear the key queue
+ * usually called when we want to empty the key queue
+ */
+export const clearKeyQueue = (keyQueueName: string) => {
+	const keyQueue = getKeyQueue(keyQueueName);
+
+	keyQueue.splice(0, keyQueue.length);
+};
```

Comment:
is slice used here to provide a mutation on a reference or could we not do some assignment? Or maybe I'm getting caught up in the naming of this function (this doesn't look feel a true "clear" operation).

Reply:
I believe splice will allow us to empty the array. To be honest I did look this up, and there was also an option to just reassign it to an empty array, but we need to make sure that other references to this variable are cleared out, hence splice. Performance wise it was documented as the same as reassignment